subxt/src/client/online_client.rs

```diff
@@ -0,0 +1,233 @@
+// Copyright 2019-2022 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+use super::{
+    OfflineClient,
+    OfflineClientT,
+};
+use crate::{
+    constants::ConstantsClient,
+    error::Error,
+    events::EventsClient,
+    rpc::{
+        Rpc,
+        RpcClient,
+        RuntimeVersion,
+    },
+    storage::StorageClient,
+    tx::TxClient,
+    Config,
+    Metadata,
+};
+use derivative::Derivative;
+use futures::future;
+use parking_lot::RwLock;
+use std::sync::Arc;
+
+/// A trait representing a client that can perform
+/// online actions.
+pub trait OnlineClientT<T: Config>: OfflineClientT<T> {
+    /// Return an RPC client that can be used to communicate with a node.
+    fn rpc(&self) -> &Rpc<T>;
+}
+
+/// A client capable of performing offline or online operations.
```

Comment:
I find these docs a bit confusing i.e, `online or offline`.

I suggest that some like: `A client capable of performing online operations but it can be converted to an offline client`

Reply:
I agree that the comment is a bit naff; it was trying to say that this client can be used in any place that requires an `OnlineClientT` or an `OfflineClientT` (and yes, it can also be converted into an `OfflineClient` if you want to limit it to only being accepted in places needing an `OfflineClientT`)